erpnext/hr/doctype/employee_checkin/employee_checkin.js

```diff
@@ -6,5 +6,9 @@ frappe.ui.form.on('Employee Checkin', {
 		if(!frm.doc.time) {
 			frm.set_value("time", frappe.datetime.now_datetime());
 		}
+	},
+	refresh: (frm) => {
+		// make log type mandatory
+		frm.set_df_property('log_type', 'reqd', frm.doc.log_type ? 0 : 1);
```

Comment:
Why not make the `log_type` field an optional field itself? Why it needs to be done this way I am not able to understand

Reply:
Yes, that's right... `log_type` was optional earlier. But in this PR https://github.com/frappe/erpnext/pull/20403 it was changed to mandatory.

But, there is a use case where `log_type` field can be left blank and this use case is only concerning the creation of employee checkin via API.

Assuming that there must be a reason why `log_type` was made mandatory in PR https://github.com/frappe/erpnext/pull/20403, the least possible change to make was to set this field to mandatory only when creating Employee Checkin from the UI.

If you find no use cases for marking this field mandatory in the UI also then we can remove this line.

Reply:
> Yes, that's right... `log_type` was optional earlier. But in this PR #20403 it was changed to mandatory.
> 
> But, there is a use case where `log_type` field can be left blank and this use case is only concerning the creation of employee checkin via API.
> 
> Assuming that there must be a reason why `log_type` was made mandatory in PR #20403, the least possible change to make was to set this field to mandatory only when creating Employee Checkin from the UI.
> 
> If you find no use cases for marking this field mandatory in the UI also then we can remove this line.

Even from the UI if the `log_type` is entered then only it will make the field mandatory. If `log_type` is not entered, it will accept without it i guess, so it acts like a optional field, this statement doesn't makes any difference

Reply:
@karthikeyan5 I think Deepesh is right just make it optional because if  log_type is entered then only it will make the field mandatory. anyway, it acts like an optional.

Reply:
done... the js validations have been removed...